README.md
This is a modified version of [cupertino_onboarding](https://pub.dev/packages/cupertino_onboarding)

* What's new feature icons support widgets, not icons only.


CHANGELOG.md
## 1.0.1

- What's New feature icons now allow any widget

## 1.0.0+1

- Readme strip

## 1.0.0

- Initial release


lib/src/widgets/whats_new_feature.dart
// Flutter imports:
import 'package:flutter/cupertino.dart';
// Project imports:
import 'package:onboard/src/constants.dart';

const double _kIconToTextSpacing = 18;
const double _kIconSize = 35;

/// Widget that represents a feature of the onboarding.
class WhatsNewFeature extends StatelessWidget {
  /// Feature's default constructor.
  const WhatsNewFeature({
    required this.title,
    required this.description,
    required this.icon,
    super.key,
  });

  /// Feature's title.
  ///
  /// Usually a Text widget.
  final Widget title;

  /// Feature's description.
  ///
  /// Usually a Text widget.
  final Widget description;

  /// Feature's icon.
  ///
  /// Color of the icon defaults to the [CupertinoTheme]'s primaryColor.
  /// Icon size defaults to 35.
  final Widget icon;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Row(
          children: [
            IconTheme(
              data: IconThemeData(
                color: CupertinoTheme.of(context).primaryColor,
                size: _kIconSize,
              ),
              child: icon,
            ),
            const SizedBox(width: _kIconToTextSpacing),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  DefaultTextStyle(
                    style: TextStyle(
                      color: kBaseTextColor.resolveFrom(context),
                      fontSize: 15,
                      fontWeight: FontWeight.w600,
                    ),
                    child: title,
                  ),
                  DefaultTextStyle(
                    style: TextStyle(
                      fontSize: 15,
                      height: 1.3,
                      letterSpacing: -0.1,
                      color: kDescriptionTextColor.resolveFrom(context),
                    ),
                    child: description,
                  ),
                ],
              ),
            ),
          ],
        ),
      ],
    );
  }
}


lib/src/pages/cupertino_onboarding_page.dart
// Flutter imports:
import 'package:flutter/cupertino.dart';
// Project imports:
import 'package:onboard/src/constants.dart';

const EdgeInsets _kOnboardingPagePadding = EdgeInsets.only(left: 35, right: 15);

const double _kTitleTopIndent = 80;

const double _kTitleToBodySpacing = 55;

/// Represents a swipeable page in the onboarding.
class CupertinoOnboardingPage extends StatelessWidget {
  /// Default constructor of the [CupertinoOnboardingPage] widget.
  const CupertinoOnboardingPage({
    required this.title,
    required this.body,
    this.bodyPadding = _kOnboardingPagePadding,
    this.titleTopIndent = _kTitleTopIndent,
    this.titleToBodySpacing = _kTitleToBodySpacing,
    this.bodyToBottomSpacing = 0,
    this.titleFlex = 3,
    super.key,
  });

  /// Title of the onboarding.
  ///
  /// It is recommended to keep it short.
  ///
  /// Defaults to Text("What's New").
  /// If another Text widget is provided, it will be
  /// defaultly styled to match the iOS 15 style onboarding.
  final Widget title;

  /// Body of the onboarding.
  final Widget body;

  /// Padding of the body.
  final EdgeInsets bodyPadding;

  /// Top indent of the title.
  ///
  /// Defaults to 80.
  final double titleTopIndent;

  /// Spacing between the title and the body.
  ///
  /// Defaults to 55.
  final double titleToBodySpacing;

  /// Spacing between the body and the bottom buttons/page indicator.
  ///
  /// Defaults to 0.
  final double bodyToBottomSpacing;

  /// Flex value of the title.
  ///
  /// Determines how much horizontal space the title takes.
  ///
  /// Defaults to 3.
  final int titleFlex;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Padding(
          padding: EdgeInsets.only(
            top: titleTopIndent,
            bottom: titleToBodySpacing,
          ),
          child: DefaultTextStyle(
            textAlign: TextAlign.center,
            style: TextStyle(
              color: kBaseTextColor.resolveFrom(context),
              fontWeight: FontWeight.w700,
              letterSpacing: -1,
              fontSize: 35,
            ),
            child: Row(
              children: [
                const Spacer(),
                Expanded(
                  flex: 3,
                  child: title,
                ),
                const Spacer(),
              ],
            ),
          ),
        ),
        Expanded(
          flex: 10,
          child: DefaultTextStyle(
            style: TextStyle(
              color: kBaseTextColor.resolveFrom(context),
            ),
            child: Padding(
              padding: bodyPadding,
              child: body,
            ),
          ),
        ),
        SizedBox(height: bodyToBottomSpacing),
      ],
    );
  }
}


lib/src/pages/whats_new_page.dart
// Flutter imports:
import 'package:flutter/cupertino.dart';
// Project imports:
import 'package:onboard/onboard.dart';

const EdgeInsets _kOnboardingPagePadding = EdgeInsets.only(left: 35, right: 15);

const double _kTitleTopIndent = 80;

const double _kTitleToBodySpacing = 55;

/// Represents an "What's new" screen in iOS 15 style.
///
/// It is possible to restyle this widget to match older iOS versions.
class WhatsNewPage extends StatelessWidget {
  /// Default constructor of the [WhatsNewPage] widget.
  ///
  /// Represents an "What's new" screen in iOS 15 style.
  /// It is possible to restyle this widget to match older iOS versions.
  WhatsNewPage({
    this.title = const Text("What's New"),
    required this.features,
    this.bodyPadding = _kOnboardingPagePadding,
    this.titleTopIndent = _kTitleTopIndent,
    this.titleToBodySpacing = _kTitleToBodySpacing,
    this.bodyToBottomSpacing = 0,
    this.titleFlex = 3,
    super.key,
  }) : assert(
          features.isNotEmpty,
          'Feature list must contain at least 1 widget.',
        );

  /// Title of the onboarding.
  ///
  /// It is recommended to keep it short.
  ///
  /// Defaults to Text("What's New").
  /// If another Text widget is provided, it will be
  /// defaultly styled to match the iOS 15 style.
  final Widget title;

  /// List of widgets that will be displayed
  /// under the title.
  ///
  /// Preferably, list of [WhatsNewFeature] widgets.
  final List<Widget> features;

  /// Padding of the body.
  final EdgeInsets bodyPadding;

  /// Top indent of the title.
  ///
  /// Defaults to 80.
  final double titleTopIndent;

  /// Spacing between the title and the body.
  ///
  /// Defaults to 55.
  final double titleToBodySpacing;

  /// Spacing between the body and the bottom buttons/page indicator.
  ///
  /// Defaults to 0.
  final double bodyToBottomSpacing;

  /// Flex value of the title.
  ///
  /// Determines how much horizontal space the title takes.
  ///
  /// Defaults to 3.
  final int titleFlex;

  @override
  Widget build(BuildContext context) {
    const _featuresSeparator = SizedBox(height: 25);

    return CupertinoOnboardingPage(
      title: title,
      bodyPadding: bodyPadding,
      titleTopIndent: titleTopIndent,
      titleToBodySpacing: titleToBodySpacing,
      bodyToBottomSpacing: bodyToBottomSpacing,
      titleFlex: titleFlex,
      body: ListView.separated(
        separatorBuilder: (_, __) => _featuresSeparator,
        itemCount: features.length,
        itemBuilder: (context, index) {
          return features[index];
        },
      ),
    );
  }
}
